src/thunderbots/software/test/world/robot.cpp

```diff
@@ -1,214 +1,328 @@
+    // id is used once https://github.com/UBC-Thunderbots/Software/issues/16i is done
+}
+
+TEST_F(RobotTest, update_state_to_predicted_state_with_future_timestamp)
+{
+    Robot robot = Robot(1, Point(1, -2), Vector(3.5, 1), Angle::ofRadians(-0.3),
+                        AngularVelocity::ofRadians(2), current_time);
+
+    robot.updateStateToPredictedState(one_second_future);
 
-    // TODO: Update this test to also test for a thrown exception once
-    // https://github.com/UBC-Thunderbots/Software/issues/16
-    // is completed. If a robot is updated using a robot with a different id, an
-    // exception should be thrown
+    EXPECT_EQ(Point(4.5, -1), robot.position());
+    EXPECT_EQ(Point(3.5, 1), robot.velocity());
+    EXPECT_EQ(Angle::ofRadians(-0.3) + Angle::ofRadians(2), robot.orientation());
+    EXPECT_EQ(AngularVelocity::ofRadians(2), robot.angularVelocity());
+    EXPECT_EQ(one_second_future, robot.lastUpdateTimestamp());
 }
 
-TEST(RobotTest, get_position_at_current_time)
+TEST_F(RobotTest, update_state_to_predicted_state_with_past_timestamp)
+{
+    // TODO: Add unit tests to check for thrown exceptions when past timestamps are used
+    // once https://github.com/UBC-Thunderbots/Software/issues/16 is done
+}
+
+TEST_F(RobotTest, get_position_at_current_time)
 {
     Robot robot = Robot(0, Point(-1.2, 3), Vector(-0.5, 2.6), Angle::quarter(),
-                        AngularVelocity::ofRadians(0.7));
+                        AngularVelocity::ofRadians(0.7), current_time);
 
     EXPECT_EQ(Point(-1.2, 3), robot.position());
 }
 
-TEST(RobotTest, get_position_at_future_time)
+TEST_F(RobotTest, get_position_at_future_time)
 {
     Robot robot = Robot(0, Point(-1.2, 3), Vector(-0.5, 2.6), Angle::quarter(),
-                        AngularVelocity::ofRadians(0.7));
+                        AngularVelocity::ofRadians(0.7), current_time);
 
-    EXPECT_EQ(Point(-1.4, 4.04), robot.estimatePositionAtFutureTime(0.4));
-    EXPECT_EQ(Point(-1.7, 5.6), robot.estimatePositionAtFutureTime(1));
-    EXPECT_EQ(Point(-2.7, 10.8), robot.estimatePositionAtFutureTime(3));
+    EXPECT_EQ(Point(-1.4, 4.04), robot.estimatePositionAtFutureTime(milliseconds(400)));
+    EXPECT_EQ(Point(-1.7, 5.6), robot.estimatePositionAtFutureTime(milliseconds(1000)));
+    EXPECT_EQ(Point(-2.7, 10.8), robot.estimatePositionAtFutureTime(milliseconds(3000)));
+}
 
+TEST_F(RobotTest, get_position_at_future_time_2)
+{
     Robot robot_other = Robot(1, Point(1, -2), Vector(3.5, 1), Angle::ofRadians(-0.3),
-                              AngularVelocity::ofRadians(2));
+                              AngularVelocity::ofRadians(2), current_time);
+
+    EXPECT_EQ(Point(2.4, -1.6),
+              robot_other.estimatePositionAtFutureTime(milliseconds(400)));
+    EXPECT_EQ(Point(4.5, -1),
+              robot_other.estimatePositionAtFutureTime(milliseconds(1000)));
+    EXPECT_EQ(Point(11.5, 1),
+              robot_other.estimatePositionAtFutureTime(milliseconds(3000)));
+}
```

Comment:
What is the difference between `get_position_at_future_time` and `get_position_at_future_time_2`? If they're testing the same thing we should remove one, and if they're testing different things they should be named as such. (I know this can be really hard, I've definitely named tests like this before, but do as I say, not as I do :grimacing:). Don't be afraid to make test names long and descriptive, and to add a quick docstring before the test explaining what it does. Looks like there a few other cases if this sort of naming (`some_test` and `some_test_2`) in this PR, so same comment goes doe all them.

Reply:
fixed